lib/ansible/modules/system/user.py

```diff
@@ -946,13 +954,22 @@ def modify_user(self):
                     new_groups = groups | set(current_groups)
                 cmd.append(','.join(new_groups))
 
-        if self.expires:
-            current_expires = time.gmtime(int(self.user_password()[1]))
+        if self.expires is not None:
+
+            current_expires = self.user_password()[1]
+
+            if self.expires < time.gmtime(0):
+                if current_expires >= 0:
+                    cmd.append('-e')
+                    cmd.append('0')
+            else:
+                # Convert days since Epoch to seconds since Epoch as struct_time
+                current_expire_date = time.gmtime(int(current_expires) * 86400)
```

Comment:
In BSD, this is stored as seconds since epoch, so no need to multiply by 86400.

```
0.0s modify_user: current_expires=2529878400
```

Reply:
that was existing code, i just converted from:
```
           # Convert days since Epoch to seconds since Epoch as struct_time
           total_seconds = int(current_expires) * 86400
           current_expires = time.gmtime(total_seconds)
```